cli/cmd/network.go

```diff
@@ -0,0 +1,171 @@
+package cmd
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/spf13/cobra"
+
+	"github.com/oasisprotocol/oasis-sdk/cli/cmd/common"
+	"github.com/oasisprotocol/oasis-sdk/cli/config"
+	"github.com/oasisprotocol/oasis-sdk/cli/table"
+)
+
+var (
+	networkCmd = &cobra.Command{
+		Use:   "network",
+		Short: "Manage network endpoints",
+	}
+
+	networkListCmd = &cobra.Command{
+		Use:     "list",
+		Aliases: []string{"ls"},
+		Short:   "List configured networks",
+		Run: func(cmd *cobra.Command, args []string) {
+			cfg := config.Global()
+			table := table.New()
+			table.SetHeader([]string{"Name", "Chain Context", "RPC"})
+
+			var output [][]string
+			for name, net := range cfg.Networks.All {
+				output = append(output, []string{
+					name,
+					net.ChainContext,
+					net.RPC,
+				})
+			}
+
+			// Sort output by name.
+			sort.Slice(output, func(i, j int) bool {
+				return output[i][0] < output[j][0]
+			})
+
+			table.AppendBulk(output)
+			table.Render()
+		},
+	}
+
+	networkAddCmd = &cobra.Command{
+		Use:   "add <name> <chain-context> <rpc-endpoint>",
+		Short: "Add a new network",
+		Args:  cobra.ExactArgs(3),
+		Run: func(cmd *cobra.Command, args []string) {
+			cfg := config.Global()
+			name, chainContext, rpc := args[0], args[1], args[2]
+
+			// Ask user for some additional parameters.
+			questions := []*survey.Question{
+				{
+					Name:   "description",
+					Prompt: &survey.Input{Message: "Description:"},
+				},
+				{
+					Name:   "symbol",
+					Prompt: &survey.Input{Message: "Denomination symbol:"},
+				},
+				{
+					Name: "decimals",
+					Prompt: &survey.Input{
+						Message: "Denomination decimal places:",
+						Default: "9",
+					},
+					Validate: survey.Required,
+				},
+			}
+			answers := struct {
+				Description string
+				Symbol      string
+				Decimals    uint8
+			}{}
+			err := survey.Ask(questions, &answers)
+			cobra.CheckErr(err)
+
+			err = cfg.Networks.Add(name, &config.Network{
```

Comment:
Could we validate user's input before asking survey questions?
I.e. running `net.Validate()` with just the given chain context and RPC so that users aren't asked those questions if we already know that the command will fail.

To prevent things like:
```
$ oasis network add dryrun 53852332637bacb61b91b6411ab4095168ba02a50be4c3f82448438826f dryrun.grpc.oasis.dev:443
? Description: bla
? Denomination symbol: DRY
? Denomination decimal places: 9
Error: malformed chain context: encoding/hex: odd length hex string
```

Reply:
Done.